libphal/utils.H

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+namespace libphal
+{
+namespace utils
+{
+
+struct BufPtr {
+       public:
+	//@brief Distructor for the object, free the allocated memory.
+	~BufPtr()
+	{
+		free(dataPtr);
+	}
+	//@brief Returns the pointer to the data
+	uint8_t **getPtr()
+	{
+		return &dataPtr;
+	}
+	// @brief Returns the stored data
+	uint8_t *getData()
+	{
+		return dataPtr;
+	}
+
+       private:
+	// The pointer to the data
+	uint8_t *dataPtr = NULL;
+};
+
+} // namespace utils
+
+using bufPtr_t = utils::BufPtr;
+
```

Comment:
This seems absolutely pointless.

Also, if you are going to create sub-namespaces  libphal::utils, I would suggest, you name the files accordingly, so it's easier to know which file defines what functionality.

I would prefer if this file is called utils_buffer.H.  Similarly, temporary_file be called utils_tempfile or similar.

Reply:
Ack